modules/ppcp-paypal-subscriptions/src/PayPalSubscriptionsModule.php

```diff
@@ -87,6 +87,36 @@ function( $product_id ) use ( $c ) {
 			12
 		);
 
+		add_filter(
+			'woocommerce_add_to_cart_validation',
+			/**
+			 * Param types removed to avoid third-party issues.
+			 *
+			 * @psalm-suppress MissingClosureParamType
+			 */
+			function ( $passed_validation, $product_id ) {
+				if ( WC()->cart->is_empty() ) {
+					return $passed_validation;
+				}
+
+				foreach ( WC()->cart->get_cart() as $cart_item_key => $cart_item ) {
+					if ( get_post_meta( $cart_item['product_id'], 'ppcp_subscription_product', true ) ) {
```

Comment:
Better use [WC Data API for product meta](https://github.com/woocommerce/woocommerce-paypal-payments/blob/trunk/modules/ppcp-paypal-subscriptions/src/SubscriptionsApiHandler.php#L141), it's more future proof, if WC decides to change how it works internally in the future we don't care as we follow the API contract.